jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py

```diff
@@ -162,3 +192,47 @@ def _get_subset_definition(self, *args):
         subset_state = subset_group.subset_state
 
         self._unpack_nested_subset(subset_state)
+
+    def vue_update_subset(self, *args):
+        if not self.is_editable:  # no-op
+            return
+
+        subset_group = [s for s in self.app.data_collection.subset_groups if
+                        s.label == self.subset_selected][0]
+        subset_state = subset_group.subset_state
```

Comment:
it might be nice to put this in the `SubsetSelect` class so its reusable.  There already is a `self.subset_select.selected_obj`, but that returns the _region_ not the _state_, and `selected_obj` is used across all selects, so we don't want to change that behavior.  Maybe this logic you have here should be `SubsetSelect.selected_subset_state`?

Reply:
I tried this but does not work (tests broke). What am I doing wrong?

```diff
--- a/jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py
+++ b/jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py
@@ -187,19 +187,14 @@ class SubsetPlugin(TemplateMixin):
         """
         self.subset_definitions = []
         self.subset_types = []
-        subset_group = [s for s in self.app.data_collection.subset_groups if
-                        s.label == self.subset_selected][0]
-        subset_state = subset_group.subset_state

-        self._unpack_nested_subset(subset_state)
+        self._unpack_nested_subset(self.subset_select.selected_subset_state)

     def vue_update_subset(self, *args):
         if not self.is_editable:  # no-op
             return

-        subset_group = [s for s in self.app.data_collection.subset_groups if
-                        s.label == self.subset_selected][0]
-        subset_state = subset_group.subset_state
+        subset_state = self.subset_select.selected_subset_state

--- a/jdaviz/core/template_mixin.py
+++ b/jdaviz/core/template_mixin.py
@@ -722,6 +722,12 @@ class SubsetSelect(BaseSelectPluginComponent):
             if match is not None:
                 return match

+    @cached_property
+    def selected_subset_state(self):
+        subset_group = [s for s in self.app.data_collection.subset_groups if
+                        s.label == self.selected][0]
+        return subset_group.subset_state
+
     def selected_min_max(self, spectrum1d):
```

Reply:
You'll have to manually handle the cache... does replacing `@cached_property` with `@property` fix the tests?

Reply:
Yes, that seems to fix it. Thanks!